### ISSUE TITLE
Indicate required and invalid fields with aria- attributes

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#2148]
See [ARIA2: Identifying a required field with the `aria-required` property](http://www.w3.org/TR/WCAG20-TECHS/aria#ARIA2).

- [x] server must add `aria-required="true"` and `aria-invalid="true"` as needed
- [x] client must add `aria-required="true"` and `aria-invalid="true"` and remove the attributes as needed

This must be done at least for:

- input
- textarea
- listbox
- dropdown

It doesn't seem like this works well for [radio buttons](http://stackoverflow.com/questions/8509481/aria-required-in-a-radio-group) or chekboxes. I put an `aria-required` on the enclosing `span`, but OS X Voice Over doesn't read it.

See also [Using the aria-required attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute).

**Comment by ebruchez** [comment]
This should also be made to work with XBL components. I notice that:

- `fr:us-phone` works
- `fr:number` and `fr:currency` don't (they use an HTML input field internally)
- `fr:tinymce` doesn't work
